src/lib.rs

```diff
@@ -1444,4 +1444,63 @@ mod tests {
         let rebuilt_root = rebuild(&proof).unwrap();
         assert_eq!(new_root, rebuilt_root);
     }
+
+    #[test]
+    fn check_payload_length_exactly_32_bytes() {
+        let mut root = Node::default();
+
+        root = insert_leaf(&mut root, &NibbleKey::from(vec![1u8; 16]), vec![1u8; 20]).unwrap();
+        assert_eq!(root.hash().len(), 32);
+        assert_eq!(
+            root.hash(),
+            vec![
+                149, 160, 25, 137, 124, 149, 98, 15, 208, 235, 90, 71, 238, 186, 81, 6, 47, 67,
+                244, 224, 134, 155, 76, 154, 130, 70, 234, 61, 0, 11, 4, 135
+            ]
+        );
+    }
+
+    #[test]
+    fn check_leaf_length_less_than_32_bytes() {
+        let mut root = Node::default();
+
+        root = insert_leaf(&mut root, &NibbleKey::from(vec![1u8; 2]), vec![1u8; 20]).unwrap();
+        assert_eq!(
+            root.hash(),
+            vec![216, 130, 32, 17, 148, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
```

Comment:
Ah this is the edge case I was mentioning. Here I think the root hash should be a hash of this leaf